### PR TITLE
Add error handling to mesh_io

### PR DESCRIPTION
**Pull request by aramSofthenge**
- Introduced exceptions in mesh read functions for better error management. 
- Updated tests to validate new behavior.

### CODE REVIEW

**Comment by aramSofthenge** [comment]
fixes #1640

**Comment by chetnieter** [comment]
jenkins test this please

File: vital/io/mesh_io.cxx [hunk 1]
```diff
@@ -104,6 +105,13 @@ read_ply2(const std::string& filename)
 {
   check_input_file(filename);
   std::ifstream input_stream(filename.c_str());
+
+  if ( ! input_stream.is_open() )
```
Comment (chetnieter) [inline vital/io/mesh_io.cxx:109]: It looks like a very similar check is already done in the `check_input_file` function: https://github.com/Kitware/kwiver/blob/75369be0a22ff9b719ce216dd05125fd4f14328e/vital/io/mesh_io.cxx#L69

File: vital/io/mesh_io.cxx [hunk 2]
```diff
@@ -263,6 +271,13 @@ read_obj(const std::string& filename)
 {
   check_input_file(filename);
   std::ifstream input_stream(filename.c_str());
+
```
Comment (chetnieter) [inline vital/io/mesh_io.cxx:274]: See comment above, file stream is checked in `check_input_file`.

**Review by chetnieter — CHANGES_REQUESTED** [review]
Looks good in general. I would suggest two additional changes. Since a file stream is opened in `check_input_file` we might as well return that or a smart pointer to that file stream rather than just opening a new one.

https://github.com/Kitware/kwiver/blob/75369be0a22ff9b719ce216dd05125fd4f14328e/vital/io/mesh_io.cxx#L68

Likewise, an output file stream is opened for no apparant reason in `check_output_file`. Perhaps we should again just return this stream and use it in the various `write_` methods. If this is done I would suggest renaming the help functions to `open_input_file` and `open_output_file` respectively.

File: vital/io/mesh_io.cxx (hunk 1 above)
Comment (aramSofthenge) [reply vital/io/mesh_io.cxx]: Thanks, I've removed the redundant checks as you suggested. The commit has been amended, and the changes are now pushed.

**Comment by chetnieter** [comment]
jenkins test this please

**Comment by chetnieter** [comment]
The linux test failure is due to a test that is unrelated to these changes and is known frequently fail due to timing out.

**Pull request merged**